File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> hashset(begin(nums), end(nums));
        int ans = 0;
        
        for(auto& num : nums)
        {
            if(hashset.find(num - 1) != hashset.end()) continue;
            
            int count = 0;
            
            while(hashset.find(num) != hashset.end())
            {
                count++;
                num++;
            }
            
            ans = max(ans, count);
        }
        
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashset = set(nums)
        ret = 0
        for num in nums:
            if num - 1 in hashset:
                continue
            count = 1
            while num + 1 in hashset:
                count += 1
                num += 1
            
            ret = max(ret, count)
        
        return ret